src/main/java/org/mqttbee/mqtt/datatypes/MqttUTF8StringImpl.java

```diff
@@ -102,7 +121,7 @@ public static MqttUTF8StringImpl from(@NotNull final String string) {
     @Nullable
     public static MqttUTF8StringImpl from(@NotNull final ByteBuf byteBuf) {
         final byte[] binary = MqttBinaryData.decode(byteBuf);
-        return (binary == null) ? null : from(binary);
+        return (binary == null || binary.length > MqttBinaryData.MAX_LENGTH) ? null : from(binary);
```

Comment:
This check is unnecessary as it is performed in `from(byte[])`
(also MqttBinaryData.decode(byteBuf) guarantees that the length is not exceeded)

Reply:
DONE